sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -126,6 +126,7 @@ class APISpec extends Specification {
     private boolean recordLiveMode
     private TestResourceNamer resourceNamer
     protected String testName
+    def containerName
```

Comment:
Instead of adding a global state for each test run why not make a new client in the tests that need it.

Reply:
As the container name and blob name generated for every tests in setup anyway, make it global does not bring extra performance issue. But have new client in test class introduced a lot more codes in each tests. I'd prefer to keep it like this. Same comments below.